File: app/src/main/java/sakaitakao/android/permissionsearch/activity/MainPreferenceActivity.java
package sakaitakao.android.permissionsearch.activity;

import sakaitakao.android.permissionsearch.R;
import android.content.Intent;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceActivity;
import android.preference.PreferenceScreen;
import android.util.Log;

/**
 * 設定画面のアクティビティ
 * 
 * @author takao
 */
public class MainPreferenceActivity extends PreferenceActivity {

	private static final String SUB_PREFERENCE_SAVED_SEARCH_COND_KEY = "saved_advanced_search_condition";
	private static final String SUB_PREFERENCE_ABOUT_KEY = "about";

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		addPreferencesFromResource(R.xml.preference);
	}

	/*
	 * (非 Javadoc)
	 * 
	 * @seeandroid.preference.PreferenceActivity#onPreferenceTreeClick(android.
	 * preference.PreferenceScreen, android.preference.Preference)
	 */
	@Override
	public boolean onPreferenceTreeClick(PreferenceScreen preferenceScreen, Preference preference) {

		Log.v("onPreferenceTreeClick", "preference.key = " + preference.getKey());

		if (SUB_PREFERENCE_SAVED_SEARCH_COND_KEY.equals(preference.getKey())) {
			startActivity(new Intent(this, SavedAdvancedSearchConditionPreferenceActivity.class));
		} else if (SUB_PREFERENCE_ABOUT_KEY.equals(preference.getKey())) {
			startActivity(new Intent(this, AboutActivity.class));
		}
		return super.onPreferenceTreeClick(preferenceScreen, preference);
	}
}

File: app/src/main/kotlin/sakaitakao/android/permissionsearch/activity/MainPreferenceActivity.kt
package sakaitakao.android.permissionsearch.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.preference.Preference
import android.preference.PreferenceFragment
import android.preference.PreferenceScreen
import android.util.Log
import sakaitakao.android.permissionsearch.R

/**
 * 設定画面のアクティビティ

 * @author takao
 */
class MainPreferenceActivity : Activity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        fragmentManager.beginTransaction().replace(android.R.id.content, MainPreferenceFragment()).commit()
    }


    /**
     * 設定画面のフラグメント
     */
    class MainPreferenceFragment : PreferenceFragment() {

        public override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)
            addPreferencesFromResource(R.xml.preference)
        }

        override fun onPreferenceTreeClick(preferenceScreen: PreferenceScreen, preference: Preference): Boolean {

            Log.v("onPreferenceTreeClick", "preference.key = " + preference.key)

            when (preference.key) {
                SUB_PREFERENCE_SAVED_SEARCH_COND_KEY ->
                    startActivity(Intent(activity, SavedAdvancedSearchConditionPreferenceActivity::class.java))

                SUB_PREFERENCE_ABOUT_KEY ->
                    startActivity(Intent(activity, AboutActivity::class.java))

                else ->
                    Log.e("onPreferenceTreeClick", "Unknown Preference key ${preference.key}")
            }
            return super.onPreferenceTreeClick(preferenceScreen, preference)
        }

        companion object {

            private val SUB_PREFERENCE_SAVED_SEARCH_COND_KEY = "saved_advanced_search_condition"
            private val SUB_PREFERENCE_ABOUT_KEY = "about"
        }
    }
}